sdks/python/test-suites/tox/py38/build.gradle

```diff
@@ -44,36 +44,48 @@ project.tasks.register("preCommitPyCoverage") {
 // e.g. pyarrow and pandas also run on PreCommit Dataframe and Coverage
 project.tasks.register("postCommitPyDep") {}
 
-// Create a test task for each supported major version of pyarrow
+// Create a test task for supported major versions of pyarrow
+// We should have a test for the lowest supported version and
+// For versions that we would like to prioritize for testing,
+// for example versions released in a timeframe of last 1-2 years.
+
 toxTask "testPy38pyarrow-3", "py38-pyarrow-3", "${posargs}"
 test.dependsOn "testPy38pyarrow-3"
 postCommitPyDep.dependsOn "testPy38pyarrow-3"
 
-toxTask "testPy38pyarrow-4", "py38-pyarrow-4", "${posargs}"
-test.dependsOn "testPy38pyarrow-4"
-postCommitPyDep.dependsOn "testPy38pyarrow-4"
+toxTask "testPy38pyarrow-9", "py38-pyarrow-9", "${posargs}"
```

Comment:
Does it make sense to test against every minor version? Could we get away with something like testing against the oldest and newest versions we support? I'm unfamiliar with how much pyarrow changes between minor releases.

Reply:
If we look at compatibility between beam and pyarrow, I think the value of testing each individual pyarrow version diminishes overtime. it might be useful to do a one-time test  when doing a large upgrade like this one, then test newer versions as they are released. Note that the "newest supported" version is also tested in regular precommit suites.

A more thorough test combination might be warranted if we are worried about interoperability of some dependencies, like pandas and pyarrow. This might be why Beam has added special compat testing for these two dependencies.